mmdet/models/dense_heads/tood_head.py

```diff
@@ -0,0 +1,763 @@
+    <https://arxiv.org/abs/2108.07755>`_.
+
+    TOOD uses Task-aligned head (T-head) and is optimized by Task Alignment
+    Learning (TAL).
+
+    Args:
+        num_dcn_on_head (int): Number of deformable convolution in the head.
+            Default: 0.
+        anchor_type (str): If set to `anchor_free`, the head will use centers
+            to regress bboxes. If set to `anchor_based`, the head will
+            regress bboxes based on anchors. Default: `anchor_free`.
+        initial_loss_cls (dict): Config of initial loss.
+
+    Example:
+        >>> self = TOODHead(11, 7)
+        >>> feats = [torch.rand(1, 7, s, s) for s in [4, 8, 16, 32, 64]]
+        >>> cls_score, bbox_pred = self.forward(feats)
+        >>> assert len(cls_score) == len(self.scales)
+    """
+
+    def __init__(self,
+                 num_dcn_on_head=0,
+                 anchor_type='anchor_free',
+                 initial_loss_cls=dict(
+                     type='FocalLossWithProb',
+                     use_sigmoid=True,
+                     gamma=2.0,
+                     alpha=0.25,
+                     loss_weight=1.0),
+                 **kwargs):
+        assert anchor_type in ['anchor_free', 'anchor_based']
+        self.num_dcn_on_head = num_dcn_on_head
+        self.anchor_type = anchor_type
+        self.epoch = 0  # which would be update in head hook!
+        super(TOODHead, self).__init__(**kwargs)
+
+        if self.train_cfg:
+            self.initial_epoch = self.train_cfg.initial_epoch
+            self.initial_assigner = build_assigner(
+                self.train_cfg.initial_assigner)
+            self.initial_loss_cls = LOSSES.build(initial_loss_cls)
+            self.assigner = self.initial_assigner
+            self.alignment_assigner = build_assigner(self.train_cfg.assigner)
+            self.alpha = self.train_cfg.alpha
+            self.beta = self.train_cfg.beta
+
+    def _init_layers(self):
+        """Initialize layers of the head."""
+        self.relu = nn.ReLU(inplace=True)
+        self.inter_convs = nn.ModuleList()
+        for i in range(self.stacked_convs):
+            if i < self.num_dcn_on_head:
+                conv_cfg = dict(type='DCNv2', deform_groups=4)
+            else:
+                conv_cfg = self.conv_cfg
+            chn = self.in_channels if i == 0 else self.feat_channels
+            self.inter_convs.append(
+                ConvModule(
+                    chn,
+                    self.feat_channels,
+                    3,
+                    stride=1,
+                    padding=1,
+                    conv_cfg=conv_cfg,
+                    norm_cfg=self.norm_cfg))
+
+        self.cls_decomp = TaskDecomposition(self.feat_channels,
+                                            self.stacked_convs,
+                                            self.stacked_convs * 8,
+                                            self.conv_cfg, self.norm_cfg)
+        self.reg_decomp = TaskDecomposition(self.feat_channels,
```

Comment:
8 Does it have a specific meaning?

Reply:
Just a hyperparam. Did not find reference in the paper.